### PR TITLE
Fix issue #3793

### PR DESCRIPTION
**Pull request by kim-fitness**

### CODE REVIEW

**Comment by TheWitness** [comment]
Jian, can you please update the changelog and append to the pull request?

**Comment by kim-fitness** [comment]
> Jian, can you please update the changelog and append to the pull request?

OK, let me do it now.

File: lib/functions.php [hunk 1]
```diff
@@ -385,18 +385,22 @@ function config_value_exists($config_name) {
...
+					foreach ($tab_array as $field_array) {
+						if (isset($field_array['items']) && isset($field_array['items'][$config_name]) && isset($field_array['items'][$config_name]['default'])) {
+							return $field_array['items'][$config_name]['default'];
+						}
 					}
 				}
 			}
 		}
+	} else {
+		return null;
```
Comment (netniV) [inline lib/functions.php:403]: The flow of the function does not return a value under certain circumstances, can you ensure that the return null is the default return method.

Comment (TheWitness) [reply lib/functions.php:403]: Agree here.

File: lib/functions.php [hunk 2]
```diff
@@ -457,14 +461,16 @@ function read_config_option($config_name, $force = false) {
 				$value = $db_setting['value'];
 			}
 
-			// Store whatever value we have in the array
-			$config_array[$config_name] = $value;
+			if (null != $value) {
```
Comment (TheWitness) [inline lib/functions.php:464]: $value != null

**Comment by TheWitness** [comment]
This pull should have gone against 1.2.x too?

**Comment by kim-fitness** [comment]
> This pull should have gone against 1.2.x too?

Yes, this pull should have been merged to 1.2.x also.

**Comment by TheWitness** [comment]
Yea, please create this pull against 1.2.x too.  Thanks!

**Comment by kim-fitness** [comment]
> Yea, please create this pull against 1.2.x too. Thanks!

Pull Request #3803 is created.

**Comment by netniV** [comment]
I'd seen this was posted to develop so wasn't so bothered by it as it will be extensively used before that is released.  Just seen that it was also applied to 1.2.x, so hopefully there's a lot of testing on that as it's changing the fundamental output of read_config_option from false to null.

**Pull request merged**